### ISSUE TITLE
[Core] Determination of parking lot congestion

### ISSUE THREAD

**Issue by AKrekhovetskyi** [AndriyKy/spot-gazer#23]
# ✅ Detection of occupied parking spaces

1. Getting parking lot data from the database
2. Get a frame from the video stream of the parking lot
3. Convert parking zone coordinates to a mask and merge with the frame (optional)
4. Detect and count the number of occupied parking spaces
5. Saving data to the database
6. Repeat steps 2-5

### Sources of video streams:

- [https://camstreamer.com/live](https://camstreamer.com/live)
- [http://www.insecam.org/en/bytag/parking/](http://www.insecam.org/en/bytag/parking/)
- [https://www.webcamtaxi.com/en/](https://www.webcamtaxi.com/en/)
- [https://www.skylinewebcams.com/webcam/](https://www.skylinewebcams.com/webcam/czech-republic/pardubice-region/lan-scaron-kroun/lanskroun.html)
- [https://www.youtube.com/watch?v=EPKWu223XEg&ab_channel=TownofCollingwood](https://www.youtube.com/watch?v=EPKWu223XEg&ab_channel=TownofCollingwood)
- [https://www.youtube.com/watch?v=IN3wNf6hKMk&ab_channel=SeeJacksonHole](https://www.youtube.com/watch?v=IN3wNf6hKMk&ab_channel=SeeJacksonHole)
- [https://www.youtube.com/watch?v=iq-CT8UQzgo&ab_channel=BrightonResort](https://www.youtube.com/watch?v=iq-CT8UQzgo&ab_channel=BrightonResort)
- [https://www.skylinewebcams.com/webcam/czech-republic/pardubice-region/lan-scaron-kroun/lanskroun.html](https://www.skylinewebcams.com/webcam/czech-republic/pardubice-region/lan-scaron-kroun/lanskroun.html)

### Datasets:

- [park Object Detection Dataset and Pre-Trained Model by baidabaida](https://universe.roboflow.com/baidabaida/park-udggu)
- [parking_lot Dataset and Pre-Trained Model by quyet Blackhole](https://universe.roboflow.com/quyet-blackhole-c5tyo/parking_lot-hnmz5/model/3)

### Tutorials:

- [Ultralytics YOLOv8 Docs](https://docs.ultralytics.com/modes/predict/) ([Predict](https://docs.ultralytics.com/modes/predict/)) (OpenCV, YOLOv8)
- [How to Perform YOLO Object Detection using OpenCV in Python](https://www.thepythoncode.com/article/yolo-object-detection-with-opencv-and-pytorch-in-python) ([GitHub](https://github.com/x4nth055/pythoncode-tutorials/blob/master/machine-learning/object-detection/yolov8_opencv.py)) (YOLOv8, OpenCV)
- [Train Yolov8 object detection on a custom dataset](https://www.youtube.com/watch?v=m9fH9OWn8YM&ab_channel=Computervisionengineer) (YOLOv8)
- [Counting vehicles in parking area-Yolov8](https://www.kaggle.com/code/rezafazel63/counting-vehicles-in-parking-area-yolov8) (YOLOv8, OpenCV)
- [How to Detect Objects in Images Using the YOLOv8 Neural Network](https://www.freecodecamp.org/news/how-to-detect-objects-in-images-using-yolov8/)
- [Detecting objects with YOLOv5, OpenCV, Python and C++](https://medium.com/mlearning-ai/detecting-objects-with-yolov5-opencv-python-and-c-c7cf13d1483c) (YOLOv5, OpenCV)
- [Object Detection using YOLO and Car Detection Implementation](https://medium.com/analytics-vidhya/object-detection-using-yolo-and-car-detection-implementation-1ec79e882875) (YOLOv3)
- [Object Detection in 10 minutes with YOLOv5 & Python](https://www.youtube.com/watch?v=fu2tfOV9vbY&ab_channel=RobMulla) (YOLOv5)
- [Car detection with YOLO Model with Keras in Python](https://sandipanweb.wordpress.com/2018/03/11/autonomous-driving-car-detection-with-yolo-in-python/) (YOLOv2, Keras, TensorFlow)
- [Vehicle Counting, Classification & Detection using OpenCV & Python](https://techvidvan.com/tutorials/opencv-vehicle-detection-classification-counting/) (OpenCV, YOLOv3)
- [Car Detection using OpenCV and Python](https://www.folio3.ai/blog/car-detection-using-opencv-and-python-within-5-minutes/) (OpenCV)
- [OpenCV Python program for Vehicle detection](https://www.geeksforgeeks.org/opencv-python-program-vehicle-detection-video-frame/) (OpenCV)

### Articles:
- [Understanding Hyper-parameter-tuning of YOLO’s](https://pub.towardsai.net/understanding-hyper-parameter-tuning-of-yolos-82aec5f6e7b3)
- [How to mask outside or inside an arbitrary shape in Python?](https://stackoverflow.com/questions/71837896/how-to-mask-outside-or-inside-an-arbitrary-shape-in-python)
- [OpenCV – Stream video to web browser/HTML page](https://pyimagesearch.com/2019/09/02/opencv-stream-video-to-web-browser-html-page/)
- [Scraping Live Stream Video With Python](https://medium.com/swlh/scraping-live-stream-video-with-python-b7154b1fedde)
